src/controllers/projects.controller.js

```diff
@@ -14,9 +15,98 @@ const del = require('del');
  */
 
 let projectService = {};
+let toolService = {};
 let herokuAuthService = {};
 let githubAuthService = {};
+let travisAuthService = {};
 const projectLocation = `${process.cwd()}/generated-projects`;
+let tooling = {};
+
+/**
+ * Updates the configs to be in a format necessary for the project to be added to the
+ * database. Adds credentials for specified tools if necessary.
+ * @param config the project's configs
+ * @param user the user creating the project
+ * @returns {Promise<*>}
+ */
+async function updateConfigs(config, user) {
+  const configs = config;
+  configs.toolingConfigurations = {};
+  for (const key of Object.keys(configs.projectConfigurations)) {
+    const toolId = configs.projectConfigurations[key];
+    if (tooling[key.toLowerCase()]) {
+      configs.toolingConfigurations[key] = await tooling[key.toLowerCase()](toolId);
+    }
+  }
+
+  configs.credentials = {};
+  if (configs.toolingConfigurations.sourceControl &&
+    configs.toolingConfigurations.sourceControl.toLowerCase() === 'github') {
+    configs.credentials.github = await githubAuthService.getGithubTokenAndUsernameForUser(user);
+  }
+  if (configs.toolingConfigurations.deployment &&
+    configs.toolingConfigurations.deployment.toLowerCase() === 'heroku') {
+    configs.credentials.heroku = await herokuAuthService.getHerokuTokenAndEmailForUser(user);
+    configs.projectConfigurations.herokuAppName = configs.projectConfigurations.projectName;
+  }
+  if (configs.toolingConfigurations.ci &&
+    configs.toolingConfigurations.ci.toLowerCase() === 'travisci') {
+    await travisAuthService.getTravisTokenForUser(user);
+  }
```

Comment:
for the `if` statements above this, can we follow a similar pattern that we used in tyr? 

Something like, 

```javascript
const services = {
  github,
  travisci,
  heroku
};
```

where those are functions. This makes it so we don't have to write a new conditional everytime we make a new service. we can just add a function.